src/screens/ParadeMapScreen/component.js

```diff
@@ -1,61 +1,90 @@
 // @flow
-import React from "react";
+import React, { Component } from "react";
+import type { ElementRef } from "react";
 import { StyleSheet, View } from "react-native";
-import MapView, { Polyline, Marker } from "react-native-maps";
-import paradeCoordinates from "../../constants/parade-coordinates";
-import { velvetColor } from "../../constants/colors";
-import Text, { scaleWithFont } from "../../components/Text";
+import Permissions from "react-native-permissions";
+import type { NavigationScreenProp, NavigationState } from "react-navigation";
+import { route, region, terminals } from "../../constants/parade-coordinates";
 import LocationCard from "./LocationCard";
+import Map from "./Map";
 
-const ParadeMapScreen = () => (
-  <View style={styles.container} testID="parade-map-screen">
-    <MapView
-      style={StyleSheet.absoluteFill}
-      region={{
-        latitude: 51.5085,
-        longitude: -0.134192,
-        latitudeDelta: 0.02,
-        longitudeDelta: 0.000000041
-      }}
-    >
-      <Polyline
-        coordinates={paradeCoordinates}
-        strokeWidth={5}
-        strokeColor={velvetColor}
-        lineJoin="bevel"
-      />
-      <Marker coordinate={{ longitude: -0.14223, latitude: 51.51616 }}>
-        <View style={styles.markerView}>
-          <Text type="xSmall" color="whiteColor">
-            A
-          </Text>
-        </View>
-      </Marker>
-      <Marker coordinate={{ longitude: -0.1265, latitude: 51.50499 }}>
-        <View style={styles.markerView}>
-          <Text type="xSmall" color="whiteColor">
-            B
-          </Text>
-        </View>
-      </Marker>
-    </MapView>
-    <LocationCard />
-  </View>
-);
+type Props = {
+  navigation: NavigationScreenProp<NavigationState>
+};
+
+type State = {
+  locationPermission: boolean
+};
+
+class ParadeMapScreen extends Component<Props, State> {
+  constructor() {
+    super();
+
+    this.state = {
+      locationPermission: false
+    };
+  }
+
+  componentDidMount() {
+    this.checkPermission();
+    this.didBlur = this.props.navigation.addListener(
+      "didBlur",
+      this.returnToParade.bind(this)
```

Comment:
You can just use `this.returnToParade` here. It is already bound to `this` because it is an arrow function on a [class propery](https://new.babeljs.io/docs/en/next/babel-plugin-proposal-class-properties.html).